Fix Azure OpenAI double ?, and JSON test issues

ralph-msft opened the pull request:
- Fixes the double ? between that path and query string in the URI that was causing the Azure OpenAI service to fail the requests with HTTP 404 errors
- Fixes some of the test issues with the downgrade from version 8.X to 4.X of System.Text.Json
- Improves the reliability of VectorStoreTests.CanAssociateFiles

Commit:
Fix double ? issues with generated URIs

Commit:
Test fixes
- Fixes JSON issues due to moving to older JSON version
- Fixes vector store test that was subject to race condition in status

azure-sdk commented:
**API change check**

API changes are not detected in this pull request.

trrwilson commented on sdk/openai/Azure.AI.OpenAI/src/Generated/Internal/ClientUriBuilder.cs:
```diff
@@ -105,11 +105,7 @@ public void AppendQuery(string name, string value, bool escape)
             Argument.AssertNotNullOrWhiteSpace(name, nameof(name));
             Argument.AssertNotNullOrWhiteSpace(value, nameof(value));
 
-            if (QueryBuilder.Length == 0)
```
Changes to the `/Generated/` folder won't persist across refreshes -- we'd need the change to happen in `autorest.csharp` for it to stay "sticky," and that'd need to be the right fix for all clients.

I'm suspicious that we're using it wrong. Where does the extra `?` character originate from?

ralph-msft replied:
This is a difference between how System.UriBuilder works in .Net Framework 4.6.2, vs how it works in .Net Core 3.1, .Net 5.0 and newer. .Net Framework 4.6.2 is where we get the double ?. If we are planning to target .Net Framework, we should address this. For now I will experiment with suppressing the generation of this file and manually implementing it with correct behaviour, or just supressing the double ? at a higher level

Commit:
Move fix for .Net Framework 4.6.2 double ? up a level.

NOTE: This only fixes the issue for Azure OpenAI. A proper fix will require updates to the autogenerated code which will in a later PR

Commit:
Merge remote-tracking branch 'origin/main' into user/ralphe/polyfill_for_older_json

Commit:
Merge with changes in main

Commit:
Merge remote-tracking branch 'origin/main' into user/ralphe/polyfill_for_older_json
